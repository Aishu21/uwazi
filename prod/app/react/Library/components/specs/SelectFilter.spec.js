"use strict";var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _SelectFilter = _interopRequireDefault(require("../SelectFilter"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('SelectFilter', () => {
  let props;

  beforeEach(() => {
    props = {
      label: 'label',
      model: 'model',
      options: ['options'] };

  });

  it('should render a text filter field with a label and passing the model', () => {
    const component = (0, _enzyme.shallow)(_react.default.createElement(_SelectFilter.default, props));
    expect(component).toMatchSnapshot();
  });

  describe('when showBoolSwitch', () => {
    it('should render the and/or bool switch', () => {
      props = {
        label: 'label',
        model: 'model',
        prefix: 'prefix',
        options: ['options'],
        showBoolSwitch: true };


      const component = (0, _enzyme.shallow)(_react.default.createElement(_SelectFilter.default, props));
      expect(component).toMatchSnapshot();
    });
  });

  describe('when sort prop is set to true', () => {
    it('should activate sorting in the MultiSelect', () => {
      props.sort = true;
      const component = (0, _enzyme.shallow)(_react.default.createElement(_SelectFilter.default, props));
      expect(component).toMatchSnapshot();
    });
  });
});