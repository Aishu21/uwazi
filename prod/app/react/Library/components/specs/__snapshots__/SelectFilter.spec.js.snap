// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFilter should render a text filter field with a label and passing the model 1`] = `
<ul
  className="search__filter is-active"
>
  <li>
    label
  </li>
  <li
    className="wide"
  >
    <MultiSelect
      model="model.values"
      onChange={[Function]}
      options={
        Array [
          "options",
        ]
      }
      optionsValue="id"
      prefix=""
      sort={false}
    />
  </li>
</ul>
`;

exports[`SelectFilter when showBoolSwitch should render the and/or bool switch 1`] = `
<ul
  className="search__filter is-active"
>
  <li>
    label
    <Switcher
      model="model.and"
      onChange={[Function]}
      prefix="prefix"
    />
  </li>
  <li
    className="wide"
  >
    <MultiSelect
      model="model.values"
      onChange={[Function]}
      options={
        Array [
          "options",
        ]
      }
      optionsValue="id"
      prefix="prefix"
      sort={false}
    />
  </li>
</ul>
`;

exports[`SelectFilter when sort prop is set to true should activate sorting in the MultiSelect 1`] = `
<ul
  className="search__filter is-active"
>
  <li>
    label
  </li>
  <li
    className="wide"
  >
    <MultiSelect
      model="model.values"
      onChange={[Function]}
      options={
        Array [
          "options",
        ]
      }
      optionsValue="id"
      prefix=""
      sort={true}
    />
  </li>
</ul>
`;
